aten/src/ATen/native/cuda/RowwiseScaledMM.cu

```diff
@@ -297,6 +297,187 @@ void f8f8bf16_rowwise_impl(
   C10_CUDA_KERNEL_LAUNCH_CHECK();
 }
 
+
+// Cutlass rowwise kernel for SM100
```

Comment:
If the kernels are mostly the same, in particular if the differences could be resolved in compile time, maybe it would be better to introduce SM arch as an additional template argument, and keep the code together.  For SM89 version of the kernel that was not possible as for SM8.x CUTLASS API is quite the different, but overall CUTLASS-based kernels code is cumbersome and repetitive, so for easier maintenance, we may prefer to avoid the duplication.

Reply:
I thought about this, but than just the arch is different, the epilogue schedule is different as well, and next we will be introducing Blackwell specific optimizations which will cause them to diverge further. The SM89 and SM90 kernels (and code paths leading to the kernels) diverged substantially, so I kept SM100 separate under the assumption that by the end of this we'll see similar amount of changes for SM100.